aws-modular/eks.tf

```diff
@@ -85,7 +85,7 @@ resource "aws_eks_node_group" "nodegroup" {
 }
 
 resource "aws_iam_role" "ng" {
-  count = local.enable_eks ? 1 : 0
+  count = local.enable_eks || var.enable_artifact_store ? 1 : 0
```

Comment:
Why does someone need Sagemaker full access if they're just spinning up a single S3 bucket?

Reply:
They do not. If someone creates just an artifact store, no additional policies get attached to the IAM role. 
In line 119,
```hcl
resource "aws_iam_role_policy_attachment" "add_roles" {
  count = local.enable_eks ? length(local.roles_to_attach) : 0

  policy_arn = local.roles_to_attach[count.index]
  role       = aws_iam_role.ng[0].name
}
```
we have the condition that makes sure that the roles are attached only when the EKS cluster is enabled. We might need Sagemaker Full Access then because people might use it as a step operator with the EKS cluster as an orchestrator.